cddl/contrib/opensolaris/lib/libdtrace/common/dt_dof.c

```diff
@@ -798,7 +798,7 @@ dtrace_dof_create(dtrace_hdl_t *dtp, dtrace_prog_t *pgp, uint_t flags)
 
 		if (i > 0) {
 			actsec = dof_add_lsect(ddo, dofa, DOF_SECT_ACTDESC,
-			    sizeof (uint64_t), 0, sizeof (dof_actdesc_t),
+			    sizeof (uintptr_t), 0, sizeof (dof_actdesc_t),
```

Comment:
This (and other changes like out) will presumably break 32-bit DTrace?

Reply:
Yes.. considering that an address will go in a `uint64_t`, and it has to have a valid tag because it is later dereferenced, which are the options to keep 32-bits compatibility?

Reply:
Essentially, `#ifdef __CHERI_PURE_CAPABILITY__ uintptr_t #else uint64_t #endif`. Using uint64_t is common in ioctls, I wonder if we have a `uintptr_or_64_t`-like type somewhere... though these changes as a whole (even with that fixed) will make freebsd32/64 break on a hybrid kernel as the kernel will be expecting capabilities, so DTrace likely needs to grow a compat layer? @brooksdavis

Reply:
Which is the best strategy, @brooksdavis ?